autogpt_platform/backend/backend/util/decorator.py

```diff
@@ -72,37 +82,115 @@ async def async_wrapper(*args: P.args, **kwargs: P.kwargs) -> Tuple[TimingInfo,
     return async_wrapper
 
 
-def error_logged(func: Callable[P, T]) -> Callable[P, T | None]:
+@overload
+def error_logged(
+    *, swallow: Literal[True]
+) -> Callable[[Callable[P, T]], Callable[P, T | None]]: ...
+
+
+@overload
+def error_logged(
+    *, swallow: Literal[False]
+) -> Callable[[Callable[P, T]], Callable[P, T]]: ...
+
+
+@overload
+def error_logged() -> Callable[[Callable[P, T]], Callable[P, T | None]]: ...
+
+
+def error_logged(
+    *, swallow: bool = True
+) -> (
+    Callable[[Callable[P, T]], Callable[P, T | None]]
+    | Callable[[Callable[P, T]], Callable[P, T]]
+):
     """
-    Decorator to suppress and log any exceptions raised by a function.
+    Decorator to log any exceptions raised by a function, with optional suppression.
+
+    Args:
+        swallow: Whether to suppress the exception (True) or re-raise it (False). Default is True.
+
+    Usage:
+        @error_logged()  # Default behavior (swallow errors)
+        @error_logged(swallow=False)  # Log and re-raise errors
     """
 
-    @functools.wraps(func)
-    def wrapper(*args: P.args, **kwargs: P.kwargs) -> T | None:
-        try:
-            return func(*args, **kwargs)
-        except Exception as e:
-            logger.exception(
-                f"Error when calling function {func.__name__} with arguments {args} {kwargs}: {e}"
-            )
+    def decorator(f: Callable[P, T]) -> Callable[P, T | None]:
+        @functools.wraps(f)
+        def wrapper(*args: P.args, **kwargs: P.kwargs) -> T | None:
+            try:
+                return f(*args, **kwargs)
+            except Exception as e:
+                logger.exception(
```

Comment:
tip: look up the difference between `logger.error` and `logger.exception`

Reply:
The exception is indeed intentional otherwise you don't know what's breaking